### DISCUSSION TITLE
Question to basic of using 2d monogame

### DISCUSSION THREAD

**Discussion by derowic** [MonoGame/MonoGame#8562] [General]
hello
is there a class in monogame that implements basic functions like movement, rotation, scaling, etc. for sprites?

I'm going to make one to make it easier to use monogame and not write all the renders by hand and here's my question is whether this approach is good or was monogame created to be used in the way described in the documentation on the website?

**Comment by ThomasFOG** [comment]
Hello,
Would you mind elaborating on what you would expect MonoGame to be doing?

The main thing is that MonoGame isn't a game engine and is intended to provide intermediate features to build your own high level features/engine such as the ones you seem to be describing. Everyone usually has its own paradigm/way of thinking about what should be an "entity" class, and it is not the role of MonoGame to go into that length. However, it may happen that MonoGame lacks some basic universal building blocks. One that comes to mind would be a tweening/easing class, for instance.

**Reply by KelsamGames** [reply]
If such features are created I would vote they exist outside of "Core MonoGame", so to speak, even if they are included in the templates by default. An addon, or series of addons, that can be removed by removing the reference to DLL(s).

**Comment by SimonDarksideJ** [comment]
You can check out the latest "How To" articles for 2D sprite rendering here:

- [2D Graphics](https://docs.monogame.net/articles/getting_to_know/howto/graphics/#2d-sprite-rendering)

Alternatively, you can preview the upcoming [2D tutorial series here](https://github.com/AristurtleDev/building-2d-games-with-monogame)

**Comment by KelsamGames** [comment]
SpriteBatch can do rotation/scaling/etc on its own, unless you are referring to having an actual updates/speed/etc process part of it. You can pass an angle, scale, color, etc to SpriteBatch when drawing your sprites. For more advanced stuff, I'd recommend looking into implementing shaders.

**Comment by rdarnold** [comment]
Yes, Monogame supports sprite rotation and scaling through use of SpriteBatch.  You should be wrapping the data and logic for how you want those features architected in your own code, but you do not need to figure out and implement a scaling algorithm or a rotation algorithm for sprites - SpriteBatch.Draw takes care of this for you.
